src/System.Security.Cryptography.Algorithms/tests/HKDFTests.cs

```diff
@@ -0,0 +1,611 @@
+            Assert.Equal("FBDB1D1B18AA6C08324B7D64B71FB76370690E1D".HexToByteArray(), prk);
+        }
+
+        [Fact]
+        public void Rfc5869ExtractSpanEmptySalt()
+        {
+            byte[] prk = new byte[20];
+            byte[] ikm = new byte[20];
+            byte[] salt = Array.Empty<byte>();
+            Assert.Equal(20, HKDF.Extract(HashAlgorithmName.SHA1, ikm, salt, prk));
+            Assert.Equal("A3CBF4A40F51A53E046F07397E52DF9286AE93A2".HexToByteArray(), prk);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(19)]
+        public void Rfc5869ExtractSpanPrkTooShort(int prkSize)
+        {
+            byte[] prk = new byte[prkSize];
+            byte[] ikm = new byte[20];
+            byte[] salt = new byte[20];
+            Assert.Throws<ArgumentException>(() => HKDF.Extract(HashAlgorithmName.SHA1, ikm, salt, prk));
+        }
+
+        [Fact]
+        public void Rfc5869ExtractSpanPrkTooLong()
+        {
+            byte[] prk = new byte[24];
+
+            for (int i = 0; i < 4; i++)
+            {
+                prk[20 + i] = (byte)(i + 5);
+            }
+
+            byte[] ikm = new byte[20];
+            byte[] salt = new byte[20];
+            Assert.Equal(20, HKDF.Extract(HashAlgorithmName.SHA1, ikm, salt, prk));
+            Assert.Equal("A3CBF4A40F51A53E046F07397E52DF9286AE93A2".HexToByteArray(), prk.Take(20).ToArray());
+
+            for (int i = 0; i < 4; i++)
+            {
+                // ensure we didn't modify anything further
+                Assert.Equal((byte)(i + 5), prk[20 + i]);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCases))]
+        public void Rfc5869ExpandByteArrayTests(Rfc5869TestCase test)
+        {
+            byte[] okm = HKDF.Expand(test.Hash, test.Prk, test.Okm.Length, test.Info);
+            Assert.Equal(test.Okm, okm);
+        }
+
+        [Fact]
+        public void Rfc5869ExpandByteArrayDefaultHash()
+        {
+            byte[] prk = new byte[20];
+            Assert.Throws<ArgumentOutOfRangeException>(() => HKDF.Expand(default(HashAlgorithmName), prk, 20, null));
+        }
+
+        [Fact]
+        public void Rfc5869ExpandByteArrayNonsensicalHash()
+        {
+            byte[] prk = new byte[20];
+            Assert.Throws<ArgumentOutOfRangeException>(() => HKDF.Expand(new HashAlgorithmName("foo"), prk, 20, null));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetRfc5869TestCases))]
+        public void Rfc5869ExpandByteArrayTamperPrkTests(Rfc5869TestCase test)
+        {
+            test.Prk[0] ^= 1;
+            byte[] okm = HKDF.Expand(test.Hash, test.Prk, test.Okm.Length, test.Info);
+            Assert.NotEqual(test.Okm, okm);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(19)]
+        public void Rfc5869ExpandByteArrayPrkTooSmall(int prkSize)
```

Comment:
It'd be good for this test to vary both the PRK size and the hash algorithm (accept it here as a string for InlineData purposes), that way we show the implementation uses the correct hash size test, not just SHA-1's 20 bytes.

Reply:
didn't notice the part about string and inline data, ended up creating method but I think both are fine